axelrod/strategies/titfortat.py

```diff
@@ -92,6 +94,39 @@ class TwoTitsForTat(Player):
     def strategy(opponent: Player) -> Action:
         return D if D in opponent.history[-2:] else C
 
+class DynamicTwoTitsForTat(Player):
+    """A player starts by cooperating and then mimics previous move by 
+    opponent with a dynamic bias based off of the opponents ratio of 
+    cooperations to total moves (so their current probability of 
+    cooperating towards cooporating regardless of the move 
+    (aka: forgiveness)."""
+
+    name = 'Dynamic Two Tits For Tat'
+    classifier = {
+        'memory_depth': 2,  # Long memory, memory-2
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+    
+    @staticmethod
+    def strategy(opponent):
+		# First move
+        if len(opponent.history) == 0:
+            # Make sure we cooporate first turn
+            return C
+        if D in opponent.history[-2:]:
+            # Probability of turning the other cheek based off of 
+            # opponent's probability of defection
+            if random.random() < (opponent.cooperations / len(opponent.history)):
```

Comment:
It's difficult for me to guess what that's from (throw the test here?) it's probably just that because you've seeded it (I'm guessing?) it's acting differently to what you wrote before? BUT because it's seeded it will **always** act the same. That's what a seed does.

For example:

```python
>>> import axelrod as axl
>>> for i in range(10):
...    axl.seed(0)  # Always the same seed
...    print(axl.random_choice(.5))
D
D
D
D
D
D
D
D
D
D
```

But if we seeded just the once outside of the loop (doing that so you can reproduce what I'm doing):

```python
>>> import axelrod as axl
>>> axl.seed(0)  # Set the seed once
>>> for i in range(10):
...    print(axl.random_choice(.5))
D
D
C
C
D
C
D
C
C
D
```